server/filestore.go

```diff
@@ -1455,6 +1473,11 @@ func (mb *msgBlock) rebuildStateLocked() (*LostStreamData, []uint64, error) {
 		if !mb.dmap.Exists(seq) {
 			mb.msgs++
 			mb.bytes += uint64(rl)
+			if ttl > 0 {
+				expires := time.Duration(ts) + (time.Second * time.Duration(ttl))
+				mb.fs.ttls.Add(seq, int64(expires))
```

Comment:
Is fs locked?

Should check ttls not nil here.

Reply:
Should we also check here if this has already expired?

Reply:
Yes I believe so re. lock.

Also we can check if already expired, although I guess if they get put into the THW, they should clean up on the next call to `expireMsgs` normally.